### PR TITLE
Fixing contributing.rst access on windows systems

### PR DESCRIPTION
**Pull request by roche-emmanuel**
This is a simple pull request to validate the changes automatically introduced with the precommit checks and fix the readthedocs automatic build accordingly.

 - [x] Add your name to `AUTHORS.md` if not there already

### CODE REVIEW

**Comment by roche-emmanuel** [comment]
Hi everyone,

In this PR, I have mainly focused on modifying the `doc/source/dev_guide/CONTRIBUTING.rst` file. Initially, this was a simple symlink pointing to `../../../CONTRIBUTING.rst`. However, when checking out the Satpy repo on a Windows machine, this symlink gets converted into a file containing only `../../../CONTRIBUTING.rst`.

At this point, if you try to run the pre-commit checks, the end-of-file checker will add a `\n` at the end of that "file," and you might then commit that change to the repo. However, for Git, this is still treated as a symlink, and it will append the `\n` character to the link source path (at least, that has been my experience).

This behavior confuses the ReadTheDocs build workflow entirely, likely because it runs on Linux and attempts to read a link source such as `../../../CONTRIBUTING.rst\n`.

The solution I’ve implemented here is to avoid using a symlink altogether and instead rely on the RST `include` directive to load the `../../../CONTRIBUTING.rst` file.

With this change, the ReadTheDocs build seems to work fine, and access to the `CONTRIBUTING.rst` file also appears to be correct based on the workflow results on this page: [https://satpy--2890.org.readthedocs.build/en/2890/dev_guide/index.html](https://satpy--2890.org.readthedocs.build/en/2890/dev_guide/index.html) (clicking on the "How to contribute" links).

So, I’m now marking this PR as ready for review. Of course, please let me know if you think there’s anything that should be changed before it can be merged. Thanks 🙏!

**Review by djhoese — CHANGES_REQUESTED** [review]
Thanks for this effort. And thanks for letting us know this causes issues on Windows. I have another suggestion based on something I've seen in other projects and something I've done in my own projects. What if we move the contents of `/CONTRIBUTING.rst` to the `doc/source/dev_guide/CONTRIBUTING.rst` file? Then replace the contents of `/CONTRIBUTING.rst` with a simple sentence like "See <URL to readthedocs contributing page> for information on contributing".

This has a couple benefits:

1. Everything rendered in the docs is *in* the `docs/` sub-directory.
2. We could potentially use all the rendering tricks provided by sphinx and our configuration of our sphinx documentation. The user should typically only see the final result on the readthedocs website.

Thoughts?

**Comment by roche-emmanuel** [comment]
Hi @djhoese ,

Thanks for your review and feedback on this PR.

Concerning your request for change above, I also thinks it's a good idea to keep all the sources for the doc generation self-contained in the doc/ folder. 

Now, from a "github perspective", I think the /CONTRIBUTING.rst file is the most important one; and this is the target that will be used at different locations on github, for instance just at the bottom of this page ;-).  And I would say that "contribution guidelines" are most of the time used by developpers while doing some work: so there is also a good chance this /CONTRIBUTING.rst file will be accessed regularly for reference. That's why I would suggest to provide an actual link to the rendered guidelines in this file, so something like that in the end:

```
Contributing Guidelines
=======================

For detailed contribution guidelines, please see our `Developer's Guide on ReadTheDocs <https://satpy.readthedocs.io/en/stable/dev_guide/index.html>`_.


.. If you're reading this file locally as a plain text, you may also
   directly refer to the file doc/source/dev_guide/CONTRIBUTING.rst for
   any unmerged/pending changes to the contribution guidelines.
``` 

**Note**: I also considered using the rst include statement there (ie. `.. include:: doc/source/dev_guide/CONTRIBUTING.rst`) but unfortunately, this rst command is ignored by the github preview, so this would now really help in a development context [plus, it would still not use any advanced rendering trick as you mentioned above]

=> I have prepared another test branch on this, updating the two CONTRIBUTING.rst files accordingly (https://github.com/roche-emmanuel/satpy/tree/moving_contributing_rst). If this sounds OK to you then I can proceed with merging that other branch in this one (Or should I wait for the review from Martin before introducing more changes on this PR anyway ? Please let me know in this case)

**Comment by mraspaud** [comment]
@roche-emmanuel you definitely don't have to wait for me to continue :)

File: README
Comment (djhoese) [inline README:1]: Just curious, what is the change in this README, file? Github just makes it look like a whitespace change or something?

Comment (roche-emmanuel) [reply README]: Hmmm, good point. I didn't investigate that one any further since it just looked like a regular file on Windows, at the end of which a new line character is added. But now checking the satpy main branch on linux I realize this is in fact another symlink (to README.rst). And again, with the auto corrections injected by the pre-commit end-of-file-fixer on Windows, this symlink will become corrupted if checked out on linux afterwards:

 on Satpy main branch (on linux):
image

on my contributing_and_rtd_build_fixes branch (on linux):
image

I also checked with cygwin on Windows, but the README file definitely appears as a regular file on that OS, here is the satpy main branch for instance:
image

So... there are a few ways to proceed from here:

1. We could again simply replace the README symlink with a **real** regular file containing a simple indication like: "Please refer to the README.rst file"
2. I have not tried that, but it might be possible to tweak the end-of-file-fixer hook to ignore some files, and in this case we could tell it to ignore the README "file"
3. I'm wondering why there is this README symlink since there is already the README.rst file next to it (as far as I know github would recognize any of README{.md,.rst,.txt} as a valid readme for a repository), so it might be a valid option to just remove the README element ?
4. You could forbid any development for satpy from Windows: this way there will be no issue due to the lack of support for symlinks... But that would be a bit drastic, would it not 😁?

Now the good thing is, I have just checked on the satpy main branch on linux and it seems those 2 files are really the only symlinks on the repository currently: 

```
kenshin@neptune:/mnt/data1/dev/projects/satpy$ find . -type l -ls
219043815      0 lrwxrwxrwx   1 kenshin  kenshin        10 août 28 20:20 ./README -> README.rst
219043787      0 lrwxrwxrwx   1 kenshin  kenshin        25 août 28 20:20 ./doc/source/dev_guide/CONTRIBUTING.rst -> ../../../CONTRIBUTING.rst
```

=> For now , I just removed the README element from the repository (so this is the option 3 above). Please let me know if you think we need to proceed differently on this point.

**Comment by djhoese** [comment]
I believe the README link was created a long time ago when either we assumed that GitHub didn't find a README with a filename extension or it didn't actually work so we *needed* it. It doesn't look necessary anymore so I'm good with the removal.

**Review by djhoese — APPROVED** [review]
Looks good to me. @mraspaud any last comments before this is merged?

**Review by mraspaud — APPROVED** [review]
LGTM!

**Pull request merged**
